core/inc/knoting/px_material.h

```diff
@@ -22,6 +22,22 @@ class PhysicsMaterial {
     void set_restitution(float restitution) { m_material->get()->setRestitution(restitution); }
     void set_px_material(float staticFriction, float dynamicFriction, float restitution);
 
+    template <class Archive>
+    void save(Archive& archive) const {
+        float dynamicFriction = m_material->get()->getDynamicFriction();
+        float staticFriction = m_material->get()->getStaticFriction();
+        float restitution = m_material->get()->getRestitution();
+        archive(CEREAL_NVP(dynamicFriction), CEREAL_NVP(staticFriction), CEREAL_NVP(restitution));
+    }
+
+    template <class Archive>
+    void load(Archive& archive) {
+        float dynamicFriction, staticFriction, restitution;
+        archive(CEREAL_NVP(dynamicFriction), CEREAL_NVP(staticFriction), CEREAL_NVP(restitution));
+        on_awake();
+        set_px_material(staticFriction, dynamicFriction, restitution);
```

Comment:
still clear what is happening, even if the style is very different. potentially the physics material properties should have an intermediate step so that the values can be reference on the component itself instead of inside of physX.

Reply:
yeah not a bad idea, not sure if maintaining double the state is necessary though? Although it does happen with any components which have to call a late on load